addon/components/base/bs-form/element.js

```diff
@@ -617,11 +639,16 @@ export default FormGroup.extend({
    * @type string
    * @private
    */
-  validation: computed('hasCustomError', 'hasErrors', 'hasWarnings', 'hasValidator', 'showValidation', 'isValidating', 'disabled', function() {
+  validation: computed('hasCustomError', 'hasErrors', 'hasCustomWarning', 'hasWarnings', 'hasValidator', 'showValidation', 'isValidating', 'disabled', function() {
     if (!this.get('showValidation') || !this.get('hasValidator') || this.get('isValidating') || this.get('disabled')) {
       return null;
+    } else if (this.get('showOwnValidation') || this.get('showAllValidations')) {
```

Comment:
The intent is to distinguish between the two possible causes for validation to be displayed

* User has triggered a `showValidationOn` state, or the component consumer has set `showOwnValidation`.
or
* Model has `customError` or `customWarning`

If `showOwnValidation` is true then I think the order of displaying messages is:

* `customError`
* The model's validation error
* `customWarning`
* The model's validation warning
* The success state

However if `showOwnValidation` is false then I think the order of validation messages is
* `customError`
* `customWarning`
* no message displayed

This isn't very explicit behaviour in the code though... perhaps if the property were called `showModelValidations` or something similar the intent would be clearer. This could be a CP observing `showOwnValidation` and `showAllValidations`, ie the same as `showValidations` but with the two `has*` properties removed.

What do you think?